utf8.h

```diff
@@ -116,7 +116,7 @@ utf8_nonnull utf8_pure utf8_weak size_t utf8cspn(const void *src,
 
 // Duplicate the utf8 string src by getting its size, malloc'ing a new buffer
 // copying over the data, and returning that. Or 0 if malloc failed.
-utf8_nonnull utf8_weak void *utf8dup(const void *src);
```

Comment:
Did you mean to remove `utf8_nonnull` ?

Reply:
Aye. `utf8dup` says it can return 0 if `malloc` fails, so it can be null and that was an oversight, doh!

Reply:
Ah cool yea. I know that `malloc` can return `NULL` but for most cases that I've heard of this, the OS will just kill the process at that point. However if you are using a custom allocator like I plan to, it is entirely possible that it keeps living on.